core/src/main/java/com/sequenceiq/cloudbreak/repository/InstanceMetaDataRepository.java

```diff
@@ -121,4 +121,7 @@
     int updateStatusIfNotTerminated(@Param("id") Long id, @Param("newInstanceStatus") InstanceStatus newInstanceStatus,
             @Param("newStatusReason") String newStatusReason);
 
+    @EntityGraph(value = "InstanceMetaData.instanceGroup", type = EntityGraph.EntityGraphType.LOAD)
```

Comment:
I am not sure that we would like to do this fetching cc @akanto

Reply:
Can you elaborate on why we shouldn't do this fetching?

I added this because the `GatewayPublicEndpointManagementService.renewCertificates` call ultimately gets the stack endpoint by calling `Stack.getPrimaryGatewayInstance()`, which in turn accesses `InstanceGroup.instanceMetadata`. I was running into errors because that field wasn't being loaded due to lazy loading, so I added this call to get it and set it manually.

I think instead of getting all instance groups and all instance metadata, I can call `InstanceGroupRepository.getPrimaryGatewayInstanceGroupByStackId` and `InstanceMetaDataRepository.getPrimaryGatewayInstanceMetadata` to populate just the objects I really need. Would that work?

Reply:
I went ahead and made the change I mentioned, and removed this new method.